src/VirtualClient/VirtualClient.Dependencies/AMDGPUDriverInstallation.cs

```diff
@@ -107,17 +141,192 @@ await this.stateManager.SaveStateAsync(nameof(AMDGPUDriverInstallation), new Sta
+            List<string> postInstallationCommands = this.PostInstallationCommands();
+
+            List<List<string>> commandsLists = new List<List<string>>
+            {
+                prerequisiteCommands,
+                installationCommands,
+                postInstallationCommands
+            };
+
+            foreach (var commandsList in commandsLists)
+            {
+                foreach (string command in commandsList)
+                {
+                    IProcessProxy process = await this.ExecuteCommandAsync(command, null, Environment.CurrentDirectory, telemetryContext, cancellationToken, true)
+                        .ConfigureAwait(false);
+
+                    if (!cancellationToken.IsCancellationRequested)
+                    {
+                        await this.LogProcessDetailsAsync(process, telemetryContext, "AMDGPUDriverInstallation", logToFile: true)
+                            .ConfigureAwait(false);
+
+                        process.ThrowIfErrored<DependencyException>(errorReason: ErrorReason.DependencyInstallationFailed);
+                    }
+                }
+            }
+
+        }
+
+        private List<string> PrerequisiteCommands(LinuxDistribution linuxDistribution)
+        {
+            List<string> commands = new List<string>();
+
+            switch (linuxDistribution)
+            {
+                case LinuxDistribution.Ubuntu:
+                    commands.Add("apt-get -yq update");
+                    commands.Add("apt-get install -yq libpci3 libpci-dev doxygen unzip cmake git");
+                    commands.Add("apt-get install -yq libnuma-dev libncurses5");
+                    commands.Add("apt-get install -yq libyaml-cpp-dev");
+                    commands.Add("apt-get -yq update");
+
+                    break;
+
+                default:
+                    break;
+            }
+
+            return commands;
+        }
+
+        private List<string> VersionSpecificInstallationCommands(LinuxDistribution linuxDistribution)
+        {
+            string installationFileName = this.LinuxInstallationFile.Split('/').Last();
+            List<string> commands = new List<string>()
+            {
+            };
+
+            switch (linuxDistribution)
+            {
+                case LinuxDistribution.Ubuntu:
+                    commands.Add($"wget {this.LinuxInstallationFile}");
+                    commands.Add($"DEBIAN_FRONTEND=noninteractive apt-get install -yq ./{installationFileName}");
+                    break;
+
+                default:
+                    break;
+            }
+
+            return commands;
+        }
+
+        private List<string> PostInstallationCommands()
+        {
+            // last command needs to be removed
+            return new List<string>
+            {
+                "amdgpu-install -y --usecase=hiplibsdk,rocm,dkms",
+                $"bash -c \"echo 'export PATH=/opt/rocm/bin${{PATH:+:${{PATH}}}}' | " +

```

Comment:
Does it work? Meaning since VC is in a shell environment, does this command update the PATH for the current shell? or you need to reload the shell environment

Reply:
We are rebooting the system after post installation commands, so reloading will happen automatically.